pkg/controller/common/settings/canonical_config.go

```diff
@@ -62,14 +62,24 @@ func MustCanonicalConfig(cfg interface{}) *CanonicalConfig {
 // MustNewSingleValue creates a new config holding a single string value.
 // Convenience constructor, will panic in the unlikely event of errors.
 func MustNewSingleValue(k string, v string) *CanonicalConfig {
-	cfg := fromConfig(ucfg.New())
+	cfg := NewCanonicalConfig()
 	err := cfg.asUCfg().SetString(k, -1, v, Options...)
 	if err != nil {
 		panic(err)
 	}
 	return cfg
 }
```

Comment:
Looks like `MustNewSingleValue ` can now just be a call to `NewSingleValue` + panic on error?